crypto/fipsmodule/ml_dsa/ml_dsa_ref/sign.c

```diff
@@ -258,15 +260,17 @@ int ml_dsa_sign_internal(ml_dsa_params *params,
   ml_dsa_polyveck_invntt_tomont(params, &h);
   ml_dsa_polyveck_sub(params, &w0, &w0, &h);
   ml_dsa_polyveck_reduce(params, &w0);
-  if(ml_dsa_polyveck_chknorm(params, &w0, params->gamma2 - params->beta)) {
+  w0_invalid = ml_dsa_polyveck_chknorm(params, &w0, params->gamma2 - params->beta);
+  if(w0_invalid) {
```

Comment:
this and the previous check (z_invalid) should not be done consecutively iiuc, because that would leak which check failed.